dspace-api/src/main/java/org/dspace/app/ldn/LDNMessageConsumer.java

```diff
@@ -116,19 +158,32 @@ private boolean evaluateFilter(Context context, Item item, String constraint) {
         return filter != null && filter.getResult(context, item);
     }
 
-    private void createLDNMessage(Context context, Item item, NotifyServiceEntity service, String pattern)
-        throws SQLException, JsonMappingException, JsonProcessingException {
-
-        LDN ldn = getLDNMessage(pattern);
+    private void createLDNMessage(Context context, Item item, NotifyServiceEntity service, String pattern,
+                                  String resubmissionID)
+            throws SQLException, JsonProcessingException {
+        // Amend current pattern name to trigger
+        // Endorsement or Review offer resubmissions: append '-resubmission' to pattern name to choose the correct
+        // LDN message template: e.g. request-endorsement-resubmission or request-review-resubmission
+        LDN ldn = (resubmissionID != null)
+                ? getLDNMessage(pattern + RESUBMISSION_SUFFIX) : getLDNMessage(pattern);
         LDNMessageEntity ldnMessage =
-            ldnMessageService.create(context, format("urn:uuid:%s", UUID.randomUUID()));
+                ldnMessageService.create(context, format("urn:uuid:%s", UUID.randomUUID()));
 
         ldnMessage.setObject(item);
         ldnMessage.setTarget(service);
         ldnMessage.setQueueStatus(LDNMessageEntity.QUEUE_STATUS_QUEUED);
         ldnMessage.setQueueTimeout(new Date());
 
-        appendGeneratedMessage(ldn, ldnMessage, pattern);
+        String actorID = null;
+        if (service.isUsesActorEmailId()) {
+            // If the service has been configured to use actorEmailId, we use the submitter's email and name
+            actorID = item.getSubmitter().getEmail();
```

Comment:
Thanks for spotting! I wonder what the best fallback option is. There is an admin email property but this may not be appropriate. An alternative may be to specifically add a new ldn config property to handle this case. Do you know the rationale behind allowing null submitters?

Reply:
I have gone for the addition of a configuration property in ldn.cfg for the fallback as it is more flexible.

Reply:
great, perfect